packages/tests/server/websockets.test.ts

```diff
@@ -1348,6 +1348,8 @@ describe('lazy mode', () => {
     const sub = client.onMessageObservable.subscribe(undefined, {
       onData: onDataMock,
     });
+    await Promise.resolve();
+    await Promise.resolve();
     expect(wsClient.connection).not.toBe(null);
 
     expect(ctx.onOpenMock).toHaveBeenCalledTimes(0);
```

Comment:
Since I used a lot of async/await and simplified the URL resolution flow to always treat the URL as a promise, we now need to wait through several event loops for this test to pass.

When I use waitFor to check that the connection is null, Vitest is unfortunately too slow, causing expect(ctx.onOpenMock).toHaveBeenCalledTimes(0); to trigger too late and leading to a test failure.

I think it’s more valuable to prioritize readability for this feature than to get this specific test to pass. When this test fails, it simply indicates that the expectation for both wsClient.connection to be null and ctx.onOpenMock not to be called within the same event loop doesn’t hold. This state does exist, but due to the asynchronous behavior, it’s now a bit harder to capture it in the test.

What do you think?